---
title: "SLIDE"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{my-vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

**SLIDE Pipeline**

The SLIDE pipeline is consisted of two steps.

1.  Parameter tuning (?)

2.  Rigorous k-fold cross validation

**Step 1: Parameter Tuning**

**1-1 Pre-Processing and Checking Input Data**

Due to the assumption-free-nature of SLIDE, there are no data modality limit to the input of SLIDE. [The key input to SLIDE are just two csv files, the data matrix x and the response vector y file.]{.underline} The x file contains your data such as single cell transcriptomics (cell by gene), or spatial proteomics (region by protein). The y file contains the responses of the data, such as severity of disease, spatial regions or clonal expansion. Since SLIDE is a regression method, when the response vector has multiple unique values (not just two classes), please make sure there are an ordinal relation between the y values.

In this tutorial, we are going to use the example Systemic Sclerosis dataset we have used in the SLIDE paper. This dataset is a human skin cell scRNA-seq dataset that we have transformed into the pseudo-bulk format. [If you have human single-cell data,]{.underline} our recommended workflow is to pseudobulk your dataset since the cell-to-cell variability is high (we are currently working on employing Metacell for SLIDE, tutorial coming soon).

If you have mouse single cell data, with the reduced cell-to-cell variability, you can consider each cell as an sample.(another tutorial)

**1-2 Parameters**

The input to SLIDE is the path to a yaml file which documents the parameters.

![the example yaml file](test_yaml_example.png)

SLIDE accepts many parameters from user input to give user as much freedom to tweak the method to their own data as possible. We have set default values to many parameters that works well with majority of the data moralities. Here, we explain what each of these parameters mean.

**x_path:** a string of the path to the data matrix (x) in the csv format [with row names and column names where each row is samples (cells, patients, regions...) and each column is a features (genes, proteins...)]{.underline}.

**y_path:** a string of the path of the response vector (y) in the csv format [with row names and column names where each row is a sample and the column would be the outcome of interest.]{.underline}

**out_path:** a string of the path of a folder to store all output files (please see below section to interpret the outputs).

**delta:** controlthe number of all latent factors. The higher the delta, the less number of latent factors will be found.

**lambda:** control the sparsity of all latent factors. The higher the lambda, the less number of features will be in a latent factor.

**spec:** control the number of significant latent factors. The higher the spec, the less number of significant latent factors will be outputted.

**y_factor:** set to false if not binary and true if binary.

**y_levels:** null if y is continuous or ordinal. If y is binary, input a list of the correct order relationship such as [0, 1] or [1, 2].

**eval_type:** the performance evaluation metric used. corr for continuous Y and auc for binary Y.

**SLIDE_iter:** the number of times to repeat the SLIDE latent factor selection algorithm. The higher the iteration, the more stable the performance would be.

**SLIDE_top_feats:** the number of top features to plot from each latent factor. If set as n, a union of the top n weighted features and top n correlated (with y) features will be outputted.

We can check what the yaml file look like by reading it in as an variable:

```{r}
library(SLIDE)
yaml_path = "/ix/djishnu/Hanxi/SLIDE/test/test.yaml"
input_params = yaml::read_yaml(yaml_path)
```

Loading data, libraries and yaml file. There are two examples we have prepared in the test folder. One of the example is of binary response data and the other of ordinal/continuous response data. Please check your yaml file at any of the website for formatting issues. (insert website here) This tutorial does ssc blah blah We first load the example yaml file which stores all of the parameters.

```{r}
library(SLIDE)
yaml_path = "/ix/djishnu/Hanxi/SLIDE/test/test.yaml"
input_params = yaml::read_yaml(yaml_path)
```

```{r}
checkDataParams(yaml_path)
```

The x_path and y_path are the path to the input data matrices and response vector in the csv format. Both row names and column names are needed. The out_path is where all the output will automatically be populated to.

Delta and Lambda are 2 key parameters for SLIDE. Delta tunes the number of latent factors, the higher the delta, the less number of latent factors will be produced. Lambda tunes the sparsity in each latent factors such as how many members each latent factors will contain. The higher the lambda, the more sparse the latent factors will be. SLIDE will automatically run the method pairwise of delta and lambda. For example, with the example yaml file, four SLIDE output will be produced with delta = 0.01 - lambda = 0.5, delta = 0.01 - lambda = 0.5...

Spec is the parameter that controls the number of significant latent factor chosen by the algorithm. The lower the spec is the

```{r setup}

```

```{r setup}
```
